lib/internal/http2/compat.js

```diff
@@ -665,6 +661,22 @@ class Http2ServerResponse extends Stream {
       encoding = 'utf8';
     }
 
+    if (this.writableEnded) {
+      if (typeof cb === 'function') {
```

Comment:
nit: combine the two if statements above into a single....

```js
if (this.writableEnded && typeof cb === 'function') {
```

Reply:
actually I think the `return this` should probably move out one step.

Reply:
See updated